src/OpenFeature.DependencyInjection/OpenFeatureBuilderExtensions.cs

```diff
@@ -353,25 +353,19 @@ public static OpenFeatureBuilder AddHandler(this OpenFeatureBuilder builder, Pro
     /// <returns></returns>
     public static OpenFeatureBuilder AddHandler(this OpenFeatureBuilder builder, string handlerName, ProviderEventTypes type, Func<IServiceProvider, EventHandlerDelegate> implementationFactory)
     {
+        if (string.IsNullOrWhiteSpace(handlerName))
+            handlerName = string.Empty;
+
         var key = string.Join(":", handlerName, type.ToString());
 
         builder.Services.PostConfigure<OpenFeatureOptions>(options => options.AddHandlerName(key));
 
-        builder.Services.TryAddKeyedSingleton(key, (serviceProvider, _) =>
+        builder.Services.AddKeyedSingleton(key, (serviceProvider, _) =>
```

Comment:
```suggestion
        builder.Services.TryAddKeyedSingleton(key, (serviceProvider, _) =>
```
It was like this. Did you revert it?

Reply:
Yeah - so I think the _desired_ behaviour is to have `AddHandler` be additive. Everytime a consumer calls the extension method the delegate they provided is added. `TryAdd` would limit this behaviour as multiple calls (with the name Handler name) would only result in one being added